internal/transport/handler_server.go

```diff
@@ -79,7 +87,9 @@ func NewServerHandlerTransport(w http.ResponseWriter, r *http.Request, stats []s
 	if v := r.Header.Get("grpc-timeout"); v != "" {
 		to, err := decodeTimeout(v)
 		if err != nil {
-			return nil, status.Errorf(codes.Internal, "malformed time-out: %v", err)
+			msg := fmt.Sprintf("malformed time-out: %v", err)
```

Comment:
Same here. Include the value in the error message.

Reply:
Looks like the `err` from `decodeTimeout` already includes the value!

https://github.com/grpc/grpc-go/blob/f54bba9af73267dcc87fcc4841dfa1e0de39d332/internal/transport/handler_server_test.go#L143-L157

The latest revision brings this back to `fmt.Sprintf("malformed time-out: %v", err)`, to avoid something confusing like `malformed time-out "tomorrow": transport: timeout unit is not recognized: "tomorrow"`.